packages/ra-core/src/auth/useCanAccessResources.ts

```diff
@@ -0,0 +1,148 @@
+import { useEffect } from 'react';
+import {
+    useQueries,
+    UseQueryOptions,
+    UseQueryResult,
+} from '@tanstack/react-query';
+import useAuthProvider from './useAuthProvider';
+import useLogoutIfAccessDenied from './useLogoutIfAccessDenied';
+import { useEvent } from '../util';
+
+/**
+ * Checks whether users can access the provided resources (resource in the RBAC sense).
+ *
+ * `useCanAccessResources` returns an object describing the state of the request:
+ *
+ * - start: { isPending: true }
+ * - success: { canAccess: true, isPending: false }
+ * - error: { error: [error from provider], isPending: false }
+ *
+ * @param {Object} params Any params you want to pass to the authProvider
+ * @param {string} params.action The action to check access for
+ * @param {string[]} params.resources The list of resources to check access for
+ * @param {Object} params.record Optional. The record to check access for
+ *
+ * @returns Return the react-query result and a canAccess property which is a map of the resources and their access status { [resource: string]: boolean }
+ *
+ * @example
+ * import { useCanAccessResources } from 'react-admin';
+ *
+ * const DeleteUserButton = ({ record }) => {
+ *     const { isPending, canAccess } = useCanAccessResources({
+ *         action: 'delete',
+ *         resources: ['users.id', 'users.name', 'users.email'],
+ *         record
+ *     });
```

Comment:
this example doesn't make sense

Reply:
Only the component name, right?

Reply:
not only: you check for the delete action and then use it to determine whether you can hows it...